verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -122,59 +95,175 @@ IndexingFactNode BuildIndexingFactsTree(
+                                                &GetRoot());
+      TreeContextVisitor::Visit(node);
+      break;
+    }
+    case NodeEnum::kModuleDeclaration: {
+      ExtractModule(node);
+      break;
+    }
+    case NodeEnum::kDataDeclaration: {
+      ExtractModuleInstantiation(node);
+      break;
+    }
+    case NodeEnum::kModulePortDeclaration: {
+      ExtractInputOutputDeclaration(node);
+      break;
+    }
+    case NodeEnum ::kNetDeclaration: {
+      ExtractNetDeclaration(node);
+      break;
+    }
+    default: {
+      TreeContextVisitor::Visit(node);
+    }
+  }
 }
 
 void IndexingFactsTreeExtractor::ExtractModule(
-    const verible::SyntaxTreeNode& node) {
+    const verible::SyntaxTreeNode& module_declaration_node) {
   IndexingNodeData module_node_data(IndexingFactType::kModule);
   IndexingFactNode module_node(module_node_data);
 
   {
-    const AutoPop p(&facts_tree_context_, &module_node);
-    ExtractModuleHeader(node);
-    ExtractModuleEnd(node);
+    const IndexingFactsTreeContext::AutoPop p(&facts_tree_context_,
+                                              &module_node);
+    ExtractModuleHeader(module_declaration_node);
+    ExtractModuleEnd(module_declaration_node);
 
-    const verible::SyntaxTreeNode& module_item_list = GetModuleItemList(node);
+    const verible::SyntaxTreeNode& module_item_list =
+        GetModuleItemList(module_declaration_node);
     Visit(module_item_list);
   }
 
-  facts_tree_context_.back()->NewChild(module_node);
+  facts_tree_context_.top().NewChild(module_node);
 }
 
 void IndexingFactsTreeExtractor::ExtractModuleHeader(
-    const verible::SyntaxTreeNode& node) {
-  const verible::TokenInfo& module_name_token = GetModuleNameToken(node);
+    const verible::SyntaxTreeNode& module_header_node) {
+  const verible::TokenInfo& module_name_token =
+      GetModuleNameToken(module_header_node);
   const Anchor module_name_anchor(module_name_token, context_.base);
 
-  facts_tree_context_.back()->Value().AppendAnchor(module_name_anchor);
+  facts_tree_context_.top().Value().AppendAnchor(module_name_anchor);
+
+  // Extracting module ports e.g. (input a, input b).
+  // Ports are treated as children of the module.
+  const verible::SyntaxTreeNode* port_list =
+      GetModulePortDeclarationList(module_header_node);
+
+  if (port_list != nullptr) {
+    const std::vector<verible::TreeSearchMatch> port_names =
+        FindAllModulePortDeclarations(*port_list);
+
+    for (const verible::TreeSearchMatch& port : port_names) {
+      const verible::SyntaxTreeLeaf* leaf =
+          GetIdentifierFromModulePortDeclaration(*port.match);
+      const Anchor port_name_anchor(leaf->get(), context_.base);
+
+      IndexingNodeData module_port(IndexingFactType::kVariableDefinition);
+      IndexingFactNode port_node(module_port);
+      port_node.Value().AppendAnchor(port_name_anchor);
+
+      facts_tree_context_.top().NewChild(port_node);
```

Comment:
Looking at the constructors available, could this be written as:

```c++
facts_tree_context_.top().NewChild({Anchor(leaf->get(), context_base)}, IndexingFactType::kVariableDefinition);
```

Totally fine as-is, I leave the style up to you, and whether or not you think this compromises clarity.